domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/PerformanceMetricsLogger.kt

```diff
@@ -0,0 +1,260 @@
+package org.oppia.android.domain.oppialogger.analytics
+
+import org.oppia.android.app.model.OppiaMetricLog
+import org.oppia.android.domain.oppialogger.ApplicationStartupListener
+import org.oppia.android.util.logging.performancemetrics.PerformanceMetricsUtils
+import org.oppia.android.util.system.OppiaClock
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class PerformanceMetricsLogger @Inject constructor(
+  private val performanceMetricsController: PerformanceMetricsController,
+  private val performanceMetricsUtils: PerformanceMetricsUtils,
+  private val oppiaClock: OppiaClock
+) : ApplicationStartupListener {
+
+  /**
+   * Use a large Long value such that the time difference based on any timestamp will be negative
+   * and thus ignored until the app records initial time during [onCreate].
+   */
+  private var firstTimestamp: Long = Long.MAX_VALUE
+
+  override fun onCreate() {
+    firstTimestamp = oppiaClock.getCurrentTimeMs()
+  }
+
+  /** Logs the apk size of the application as a performance metric. */
+  fun logApkSize() {
+    performanceMetricsController.logLowPriorityMetricEvent(
+      oppiaClock.getCurrentTimeMs(),
+      OppiaMetricLog.CurrentScreen.HOME_SCREEN,
```

Comment:
Is this actually the correct screen to use? Ditto for the others.

Reply:
Actually, I think it'd be better if we let the callers set the current screen. Does that make sense?

Reply:
Yep that seems fine. I just wanted to make sure that we weren't arbitrarily defaulting it here.